martin-core/src/lib.rs

```diff
@@ -1,6 +1,6 @@
 #![doc = include_str!("../README.md")]
 #![forbid(unsafe_code)]
-#![deny(missing_docs)]
+#![forbid(missing_docs)]
```

Comment:
these configurations should probably be done as part of Cargo.toml.  Also, never use `forbid` for things like this -- it should be a WARNING, and that warning should be treated as error in the CI, but we do not want to require each new function to have documentation while you are developing.